.github/workflows/test.yml

```diff
@@ -18,23 +18,10 @@ jobs:
     - uses: actions/checkout@v2
 
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
-
-    - name: Get pip cache dir
-      id: pip-cache
-      run: |
-        echo "::set-output name=dir::$(pip cache dir)"
-
-    - name: Cache
-      uses: actions/cache@v2
-      with:
-        path: ${{ steps.pip-cache.outputs.dir }}
-        key:
-          ${{ matrix.python-version }}-v1-${{ hashFiles('**/setup.py') }}
-        restore-keys: |
-          ${{ matrix.python-version }}-v1-
+        cache: 'pip'  # caching pip dependencies
 
     - name: Install dependencies
       run: |
```

Comment:
We may also skip the Pip version check, relying on GitHub to ship a fairly recent, continuously updated version. E.g.

```bash
python -m pip install tox-gh-actions wheel --disable-pip-version-check
```